crates/virtio-queue/src/queue.rs

```diff
@@ -1238,18 +1248,8 @@ mod tests {
         // Decrement `idx` which should be forbidden. We don't enforce this thing, but we should
         // test that we don't panic in case the driver decrements it.
         vq.avail().idx().store(u16::to_le(1));
-
-        loop {
-            q.disable_notification(mem).unwrap();
-
-            while let Some(_chain) = q.iter(mem).unwrap().next() {
-                // In a real use case, we would do something with the chain here.
-            }
-
-            if !q.enable_notification(mem).unwrap() {
-                break;
-            }
-        }
+        // Invalid available ring index
```

Comment:
We should add some additional tests here:
1. A regression that fails before the fix and passes afterwards. We can write a unit test for the scenario described by Rob where the 
2. A test to check that you can still handle exactly queue size descriptor chains.

Reply:
A new unit test is added from a separate commit. Anything I am missing, please let me know. Thank you!